### DISCUSSION TITLE
Using Environment Variables on Vercel.com

### DISCUSSION THREAD

**Discussion by ECWireless** [vercel/vercel#4663] [Help] [answered]
I've deployed a node.js app on vercel, but am unable to use the environment variables I've set within my project settings. I'd assume all I have to do is write process.env.<MY_VARIABLE>, but this isn't working.

**Comment by styfle** [comment, marked as answer]
There are 3 possible environments in the [Environment Variable UI](https://vercel.com/blog/environment-variables-ui):

- Production
- Preview
- Development

If you deployed using the default git branch (for example, master) then that is a Production deployment.

If you deployed any other git branch, then that is a Preview deployment.

Local development with `vc dev` can use the development environment by pulling the variables into a file with `vc env pull`.

https://vercel.com/docs/v2/build-step#environment-variables

**Reply by ECWireless** [reply]
I've tried adding the variable to all 3 environments and none work. My usage in the reactjs website is by writing "process.env.<MY_VARIABLE>." Is this the correct format?

**Reply by waptik** [reply]
> I've tried adding the variable to all 3 environments and none work. My usage in the reactjs website is by writing "process.env.<MY_VARIABLE>." Is this the correct format?

I don't know why it doesn't work for you but it does for me, both production and preview. For dev, i use .env locally

**Reply by ECWireless** [reply]
Hmmm, okay, I'll keep tinkering with it. Thanks anyway!
